layers.go

```diff
@@ -1442,10 +1442,12 @@ func (r *layerStore) create(id string, parentLayer *Layer, names []string, mount
 		}
 	}
 
-	delete(layer.Flags, incompleteFlag)
-	if err = r.saveFor(layer); err != nil {
-		cleanupFailureContext = "saving finished layer metadata"
-		return nil, -1, err
+	if !partial {
```

Comment:
If `partial`, the code below `applyDiffWithOptions() would have updated r.bycompressedsum` modifies `layer` data and we wouldn’t save it here.

I think that might be correct enough right now (because those fields aren’t set) but it’s hard for me to track such indirect effects — the code might modify more fields in the future.